odl/discr/lp_discr.py

```diff
@@ -427,6 +511,73 @@ def cell_volume(self):
         """Cell volume of an underlying regular grid."""
         return self.space.cell_volume
 
+    @property
+    def real(self):
+        """Real part of this element."""
+        return self.space.as_real_space().element(self.asarray().real)
```

Comment:
Caching now, but it's troublesome to do it during init. For example, in `FnBase`, the weighting is not yet defined when needed. I decided to store the trivial one (`self`) immediately and the calculated counterpart in the first invocation of the corresponding method.